frontend/libs/studio-content-library/src/ContentLibrary/LibraryBody/pages/CodeList/CodeLists/EditCodeList/EditCodeList.tsx

```diff
@@ -0,0 +1,69 @@
+import type { CodeList, CodeListEditorTexts } from '@studio/components';
+import { StudioCodeListEditor, StudioToggleableTextfield } from '@studio/components';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import type { CodeListWithMetadata } from '../../CodeList';
+import { useOptionListEditorTexts } from '../../hooks/useCodeListEditorTexts';
+import { KeyVerticalIcon } from '@studio/icons';
+import { FileNameUtils } from '@studio/pure-functions';
+import { useInputCodeListNameErrorMessage } from '../../hooks/useInputCodeListNameErrorMessage';
+
+export type EditCodeListProps = {
+  codeList: CodeListWithMetadata;
+  onUpdateCodeListId: (codeListId: string, newCodeListId: string) => void;
+  onUpdateCodeList: (updatedCodeList: CodeListWithMetadata) => void;
+  codeListNames: string[];
+};
+
+export function EditCodeList({
+  codeList,
+  onUpdateCodeListId,
+  onUpdateCodeList,
+  codeListNames,
+}: EditCodeListProps): React.ReactElement {
+  const { t } = useTranslation();
+  const editorTexts: CodeListEditorTexts = useOptionListEditorTexts();
+  const getInvalidInputFileNameErrorMessage = useInputCodeListNameErrorMessage();
+
+  const handleUpdateCodeListId = (newCodeListId: string) => {
+    if (newCodeListId !== codeList.title) onUpdateCodeListId(codeList.title, newCodeListId);
+  };
+
+  const handleUpdateCodeList = (updatedCodeList: CodeList) => {
+    onUpdateCodeList({ title: codeList.title, codeList: updatedCodeList });
+  };
+
+  const handleValidateCodeListId = (newCodeListId: string) => {
+    const fileNameError = FileNameUtils.findFileNameError(newCodeListId, codeListNames);
+    return getInvalidInputFileNameErrorMessage(fileNameError);
+  };
```

Comment:
See above, and I think we should pass a regex to the validation method, it will lead to changes in `useInputCodeListNameErrorMessage.tsx` for handling the new error type.
```suggestion
  const handleValidateCodeListId = (newCodeListId: string) => {
    const fileNameRegex = /^[a-zA-Z0-9_.\s]+(?<![._\s])$/;
    const fileNameError = FileNameUtils.findFileNameError(
      newCodeListId,
      invalidCodeListNames,
      fileNameRegex,
    );
    
    return getInvalidInputFileNameErrorMessage(fileNameError);
  };
```